src/models.py

```diff
@@ -106,6 +108,16 @@ def validation_step(self, batch: Tensor, batch_idx: int):
             on_epoch=True,
             sync_dist=True,
             add_dataloader_idx=True)
+        
+        self.log(
+            name="val_perplexity", 
+            value=perplexity, 
+            prog_bar=True,
+            logger=True, 
+            on_step=False, 
+            on_epoch=True,
+            sync_dist=True,
+            add_dataloader_idx=True)
 
         return loss
```

Comment:
See comment below about returning loss and perplexity. Probably not needed as this is for training, but it still looks somewhat odd.